Example-Crypto-REXX/RXCRYPT1.rexx

```diff
@@ -0,0 +1,2428 @@
+ /*                                                            */
+ /* NOTE: if you area confident the CPC has LPARs but an empty */
+ /* list is returned, or an LPAR you expect to be in the list  */
+ /* is not there, ensure you have the appropriate access level */
+ /* to the FACILITY Class profile for the the expected LPAR(s) */
+ /* in addition to verifying firmware security configurations. */
+ /**************************************************************/
+ LPARs = JSON_getArrayDim( LPARArray )
+ if LPARs = -1 then
+    return fatalError( '** Unable to retrieve number of array entries **' )
+ else if LPARs = 0 then
+    return fatalError( '** empty list of LPARs returned **' )
+
+ /********************************************************************/
+ /* Traverse the LPARs array to populate LPARsList.                  */
+ /* We use the REXX (1-based) idiom  but adjust for the differing    */
+ /* (0-based) idiom of the toolkit.                                  */
+ /********************************************************************/
+ say 'Processing information for '||LPARs||' LPAR(s) returned from LIST LPARs'
+ drop LPARsList.
+ LPARsList.0 = LPARs
+ do i = 1 to LPARs
+    nextEntryHandle = JSON_getArrayEntry(LPARArray,i-1)
+    LPARsList.i.uri=JSON_findValue(nextEntryHandle,"object-uri",,
+                     HWTJ_STRING_TYPE)
+    if foundJSONValue(LPARsList.i.uri) = FALSE then
+      do
+        errMsg = 'Failed to obtain uri for LPAR entry ('||i||')'
+        return fatalError(errMsg)
+      end
+
+    LPARsList.i.targetName=JSON_findValue(nextEntryHandle,"target-name",,
+                     HWTJ_STRING_TYPE)
+    if foundJSONValue(LPARsList.i.targetName) = FALSE then
+      do
+        errMsg = 'Failed to obtain target name for LPAR entry ('||i||')'
+        return fatalError(errMsg)
+      end
+
+    LPARsList.i.name=JSON_findValue(nextEntryHandle,"name",HWTJ_STRING_TYPE)
+    if foundJSONValue(LPARsList.i.name) = FALSE then
+      do
+        errMsg = 'Failed to obtain name for LPAR entry ('||i||')'
+        return fataError(errMsg)
+      end
+
+    LPARsList.i.status = JSON_findValue(nextEntryHandle,"status",,
+                     HWTJ_STRING_TYPE)
+    if foundJSONValue(LPARsList.i.status) = FALSE then
+      do
+        errMsg = 'Failed to obtain status for LPAR entry ('||i||')'
+        return fatalError(errMsg)
+      end
+ end /* endloop thru the JSON LPARs array */
+
+return 0
+
+/*******************************************************/
+/* Function:  QueryLPAR                                */
+/*                                                     */
+/* Retrieve LPAR attributes identified by              */
+/* LPARAttribute stem variable                         */
+/*                                                     */
+/* Return 0 if successful, otherwise return -1         */
+/*******************************************************/
+QueryLPAR:
```

Comment:
please rename the method and the associated vars, you're querying image activation profile uri

Reply:
Done